### ISSUE TITLE
overlapping labels

### ISSUE THREAD

**Issue by Hadionly** [Abhinandan-Kushwaha/react-native-gifted-charts#801]
Im experiencing the following overlapping:
![image](https://github.com/user-attachments/assets/87ba8d56-bec4-4d2b-a29c-a04633dc5ad6)
![image](https://github.com/user-attachments/assets/846af1ea-8d4a-43c5-9683-1d81c1b734f8)

code snippet:
```
  const data1 = [
    {
      color: "#FFFFE0",
      dataPoint: "110k",
      frontColor: "#FFFFE0",
      label: "EAND",
      value: 3.985507246376811,
    },
    {
      color: "#FF6347",
      dataPoint: "70k",
      frontColor: "#FF6347",
      label: "FAB",
      value: 2.536231884057971,
    },
    {
      color: "lightgreen",
      dataPoint: "400k",
      frontColor: "lightgreen",
      label: "ADCB",
      value: 14.492753623188406,
    },
    {
      color: "rgb(255,165,0)",
      dataPoint: "330k",
      frontColor: "rgb(255,165,0)",
      label: "QHOLDING",
      value: 11.956521739130435,
    },
    {
      color: "rgb(234,134,453)",
      dataPoint: "2M",
      frontColor: "rgb(234,134,453)",
      label: "TAQA",
      value: 67.02898550724638,
    },
  ];
  const data2 = [
    {
      color: "#FFFFE0",
      dataPoint: "10k",
      dataPoint1: "110k AED",
      frontColor: "#FFFFE0",
      label: "EAND",
      value: 1.3888888888888888,
    },
    {
      color: "#FF6347",
      dataPoint: "10k",
      dataPoint1: "70k AED",
      frontColor: "#FF6347",
      label: "FAB",
      value: 1.3888888888888888,
    },
    {
      color: "lightgreen",
      dataPoint: "100k",
      dataPoint1: "400k AED",
      frontColor: "lightgreen",
      label: "ADCB",
      value: 13.88888888888889,
    },
    {
      color: "rgb(255,165,0)",
      dataPoint: "100k",
      dataPoint1: "330k AED",
      frontColor: "rgb(255,165,0)",
      label: "QHOLDING",
      value: 13.88888888888889,
    },
    {
      color: "rgb(234,134,453)",
      dataPoint: "500k",
      dataPoint1: "2M AED",
      frontColor: "rgb(234,134,453)",
      label: "TAQA",
      value: 69.44444444444444,
    },
  ];
  
          <PieChart
            extraRadius={s(100)}
            data={data}
            showExternalLabels
            externalLabelComponent={(item) => (
              <SvgText fontSize={ms(10)} fill={item?.color} fontWeight={"300"}>
                {/* {item?.dataPoint} */}
                {item?.dataPoint}/{item?.value.toFixed(0)}%
              </SvgText>
            )}
            labelLineConfig={{
              color: COLORS.white,
              labelComponentWidth: s(40),
              length: 4,
            }}

          />
```

**Comment by Hadionly** [comment]
Solved it using the following sort that keeps everything away from each other:

  const wiggleSort = (arr) => {
    // Sort the array based on the 'value' key in ascending order
    arr.sort((a, b) => a.value - b.value);

    // Create a new array to store the result
    const result = new Array(arr.length);

    // Fill even indices with objects having smaller values
    let evenIndex = 0;
    for (let i = 0; i < arr.length; i += 2) {
      result[i] = arr[evenIndex];
      evenIndex++;
    }

    // Fill odd indices with objects having larger values
    let oddIndex = arr.length - 1;
    for (let i = 1; i < arr.length; i += 2) {
      result[i] = arr[oddIndex];
      oddIndex--;
    }

    return result;
  };

**Comment by Abhinandan-Kushwaha** [comment]
Good solution @Hadionly 
I have also made a fix in the library, through which overlapping of external labels will be avoided. Usually the overlap was happening near the poles (top or bottom). So, now the library renders the external labels for sections near the poles directly above (or below) them instead of drawing a line towards left or right and then rendering the label.

This behaviour of automatically avoiding the overlapping of external labels can be disabled by setting the property `avoidOverlappingOfLabels` to`false` inside the `labelLineConfig` prop.

This is available from versions `1.4.31` onwards. Please use the latest version of the library.
